torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -343,6 +343,42 @@ class CudaKernelGenerator : private kir::IrVisitor {
     return expr.str();
   }
 
+  // If one argument is a tensorview and the other is a scalar, make sure we
+  // cast the scalar to the tensorview type
+  std::string scalarCast(kir::Val* lhs, kir::Val* rhs) {
```

Comment:
@shmsong could you please review this section to make sure it's consistent with pytorch type promotion?

Reply:
This part is consistent when the `TensorView` has floating point. https://github.com/csarofeen/pytorch/blob/20_12_3_devel/aten/src/ATen/native/TypeProperties.cpp#L59-L73

@rdspring1 , just wanted to mention that when `TensorView` is integer and `Scalar` is floating point the eager mode will cast the `TensorView` to floating point. Not sure if this case is relevant/reachable here. 

Our duplication of the type combine logic is here if it helps. 
https://github.com/csarofeen/pytorch/blob/20_12_3_devel/aten/src/ATen/native/TypeProperties.cpp#L59-L73